lib/routes/kelownacapnews/news.ts

```diff
@@ -0,0 +1,91 @@
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/:type',
+    categories: ['new-media'],
+    example: '/kelownacapnews/local-news',
+    parameters: { type: 'Type of news' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['www.kelownacapnews.com/:type'],
+            target: '/:type',
+        },
+    ],
+    name: 'News',
+    maintainers: ['hualiong'],
+    url: 'www.kelownacapnews.com',
+    description: `\`type\` is as follows:
+  
+| News type     | Value         | News type    | Value        |
+| ------------- | ------------- | ------------ | ------------ |
+| News          | news          | Sports       | sports       |
+| Local News    | local-news    | Business     | business     |
+| Canadian News | national-news | Trending Now | trending-now |
+| World News    | world-news    | Opinion      | opinion      |
+| Entertainment | entertainment |              |              |`,
+    handler: async (ctx) => {
+        const type = ctx.req.param('type');
+        const baseURL = 'https://www.kelownacapnews.com';
+
+        const response = await ofetch(`${baseURL}/${type}`);
+        const $ = load(response);
+
+        const list = $('.media')
+            .toArray()
+            .map((item): DataItem => {
+                const a = $(item);
+                return {
+                    title: a.find('.media-heading').text(),
+                    pubDate: parseDate(a.find('.media-links time').attr('datetime')!),
+                    link: baseURL + a.attr('href'),
+                };
+            });
+
+        const items = await Promise.all(
+            list.map((item) =>
+                cache.tryGet(item.link!, async () => {
+                    const response = await ofetch(item.link!);
+                    const $ = load(response);
+
+                    const details = $('.details-file');
+                    const image = details.length ? details.find('img').attr('src') : $('#sliderImgs .tablist-item .galleryWrap').toArray();
+
+                    const byline = $('.details-byline');
+                    const profileTitle = byline.find('.profile-title');
+                    if (profileTitle.length) {
+                        item.author = profileTitle.find('a').text();
+                    }
+                    let label = '';
+                    if (typeof image === 'string') {
+                        label = `<img src='${image}' />`;
+                    } else {
+                        for (const e of image!) {
+                            label += `<img src='${$(e).data('src')}' />`;
+                        }
+                    }
```

Comment:
https://github.com/DIYgod/RSSHub/pull/15945#discussion_r1648104815 https://github.com/DIYgod/RSSHub/pull/15945#discussion_r1648106783
captions are still missing

Reply:
Uh... You mean the `<img>` tag lacks the `alt` attribute? Or does it mean something else?🤔

Reply:
![image](https://github.com/DIYgod/RSSHub/assets/11386903/d50929a2-bbe1-4bb6-9ae0-709d9b9cd548)
https://developer.mozilla.org/docs/Web/HTML/Element/figcaption

Reply:
OK